samcli/cli/formatters.py

```diff
@@ -0,0 +1,66 @@
+"""
+Click Help Formatter Classes that are customized for the root command.
+"""
+from contextlib import contextmanager
+from typing import Sequence, Iterator, Optional
+
+from click import style, HelpFormatter
+
+from samcli.cli.row_modifiers import RowDefinition, BaseLineRowModifier
+from samcli.cli.root.command_list import SAM_CLI_COMMANDS
+
+
+class RootCommandHelpTextFormatter(HelpFormatter):
+
+    # Picked an additive constant that gives an aesthetically pleasing look.
+    ADDITIVE_JUSTIFICATION = 10
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # NOTE(sriram-mv): Do not left justify for more than half the width.
+        self.left_justification_length = min(
+            max([len(command) for command, _ in SAM_CLI_COMMANDS.items()]) + self.ADDITIVE_JUSTIFICATION,
```

Comment:
I'm probably missing something, but not completely following the logic here for determining the justification length.

Reply:
We want to add justification length based on the command that has the longest length. eg: `sam sync` is shorter than `sam pipeline`, so the command help text should start after the longest command name, otherwise the alignment would not look right.